internal/proxy/influxdb/influxdb.go

```diff
@@ -70,23 +106,209 @@ func (c Client) OriginName() string {
 }
 
 // DeriveCacheKey ...
-func (c Client) DeriveCacheKey(string, url.Values, string, string) string {
-	return ""
+func (c Client) DeriveCacheKey(path string, params url.Values, prefix string, extra string) string {
+	k := path
+	// if we have a prefix, set it up
+	if len(prefix) > 0 {
+		k += prefix
+	}
+
+	if p, ok := params[upQuery]; ok {
+		k += p[0]
+	}
+
+	if len(extra) > 0 {
+		k += extra
+	}
+	return md5.Checksum(k)
+}
+
+func (c *Client) ProxyHandler(w http.ResponseWriter, r *http.Request) {
+	proxy.ProxyRequest(proxy.NewRequest(c.Name, proxy.OtInfluxDb, "APIProxyHandler", r.Method, c.BuildUpstreamURL(r), r.Header, r), w)
+}
+
+func (c *Client) QueryRangeHandler(w http.ResponseWriter, r *http.Request) {
+}
+
+func (c Client) QueryHandler(w http.ResponseWriter, r *http.Request) {
+	u := c.BuildUpstreamURL(r)
+	proxy.ObjectProxyCacheRequest(proxy.NewRequest(c.Name, proxy.OtInfluxDb, "QueryHandler", r.Method, u, r.Header, r), w, &c, c.Cache, 30, false, false)
+}
+
+func getTimeValueForQueriesWithoutNow(timeParsed []string) int64 {
+	suffix := strings.SplitAfterN(timeParsed[0], " ", 2)
+	timeWithoutOperator := strings.TrimSpace(suffix[1])
+	timeWithoutOperator = timeWithoutOperator[0:]
+	unit := timeWithoutOperator[len(timeWithoutOperator)-1]
+	var multiplier = time.Nanosecond
+
+	switch unit {
+	case 'u':
+	case 'µ':
+		multiplier = time.Microsecond
+	case 'w':
+		multiplier = 24 * 7 * time.Hour
+	case 'd':
+		multiplier = 24 * time.Hour
+	case 'h':
+		multiplier = time.Hour
+	case 'm':
+		multiplier = time.Minute
+	case 's':
+		multiplier = time.Second
+	default:
+		if timeWithoutOperator[len(timeWithoutOperator)-2] == 'm' {
+			multiplier = time.Millisecond
+		} else {
+			multiplier = time.Nanosecond
+		}
+	}
+	re := regexp.MustCompile("[0-9]+")
+	number := re.FindAllString(timeWithoutOperator, -1)
+	numValue, err := (strconv.ParseInt(number[0], 10, 64))
+	if (err != nil) {
+		panic(err)
+	}
+	timeValue := numValue * multiplier.Nanoseconds()
+	return timeValue
+}
+
+func getTimeValueForQueriesWithNow(timeParsed []string) (int64, string) {
+	suffix := strings.SplitAfterN(timeParsed[0], "now()", 2)
+	timeWithOperator := strings.TrimSpace(suffix[1])
+	timeWithoutOperator := timeWithOperator[2:]
+	unit := timeWithoutOperator[len(timeWithoutOperator)-1]
+	var multiplier = time.Nanosecond
+	switch unit {
+	case 'y':
+		multiplier = 365 * 24 * time.Hour
+	case 'w':
+		multiplier = 24 * 7 * time.Hour
+	case 'd':
+		multiplier = 24 * time.Hour
+	case 'h':
+		multiplier = time.Hour
+	case 'm':
+		multiplier = time.Minute
+	case 's':
+		multiplier = time.Second
+	default:
+		multiplier = time.Nanosecond
```

Comment:
since the value is already set to time.Nanosecond at instantiation, this switch will unnecessarily set the value of `multiplier` a second time. Changing the instantiation to be `var multiplier time.Duration` will eliminate an unnecessary double-set of the variable.